fv3core/stencils/dyn_core.py

```diff
@@ -179,381 +197,415 @@ def dyncore_temporaries(shape):
+                        domain=grid.domain_shape_full(add=(0, 0, 1)),
                     )
                 else:
-                    reqs["pkc_quantity"] = comm.start_halo_update(
-                        state.pkc_quantity, n_points=utils.halo
+                    copy_stencil(
+                        state.zh,
+                        state.gz,
+                        origin=grid.full_origin(),
+                        domain=grid.domain_shape_full(add=(0, 0, 1)),
                     )
-            if remap_step:
-                pe_halo.compute(state.pe, state.delp, state.ptop)
-            if spec.namelist.use_logp:
-                raise Exception("unimplemented namelist option use_logp=True")
-            else:
-                pk3_halo.compute(state.pk3, state.delp, state.ptop, akap)
-        if not hydrostatic:
-            if global_config.get_do_halo_exchange():
-                reqs["zh_quantity"].wait()
-                if grid.npx != grid.npy:
-                    reqs["pkc_quantity"].wait()
-            basic.multiply_constant(
-                state.zh,
-                state.gz,
-                constants.GRAV,
-                origin=(grid.is_ - 2, grid.js - 2, 0),
-                domain=(grid.nic + 4, grid.njc + 4, grid.npz + 1),
-            )
-            if grid.npx == grid.npy and global_config.get_do_halo_exchange():
-                reqs["pkc_quantity"].wait()
-            if spec.namelist.beta != 0:
-                raise Exception(
-                    "Unimplemented namelist option -- we only support beta=0"
+            if not hydrostatic:
+                updatedzc.compute(
+                    state.dp_ref, state.zs, state.ut, state.vt, state.gz, state.ws3, dt2
                 )
-            nonhydrostatic_pressure(
-                state.u,
-                state.v,
+                riem_solver_c.compute(
+                    ms,
+                    dt2,
+                    akap,
+                    state.cappa,
+                    state.ptop,
+                    state.phis,
+                    state.omga,
+                    state.ptc,
+                    state.q_con,
+                    state.delpc,
+                    state.gz,
+                    state.pkc,
+                    state.ws3,
+                )
+
+            self._p_grad_c(
+                grid.rdxc,
+                grid.rdyc,
+                state.uc,
+                state.vc,
+                state.delpc,
                 state.pkc,
                 state.gz,
-                state.pk3,
-                state.delp,
-                dt,
-                state.ptop,
-                akap,
+                dt2,
             )
-
-        if spec.namelist.rf_fast:
-            # TODO: Pass through ks, or remove, inconsistent representation vs Fortran.
-            ray_fast.compute(
+            if self.do_halo_exchange:
+                reqc_vector = self.comm.start_vector_halo_update(
```

Comment:
`reqc` made me do a double take, can you simply name this `req_vector` (I think this is the important part of the intent), or more verbosely `req_vector_c_grid`? even `req` is fine, here, since you're waiting on the request immediately.